ci/lib.sh

```diff
@@ -49,43 +49,87 @@ arch() {
   esac
 }
 
+# A helper function to get the artifacts url
+# Takes 1 argument:
+# $1 = environment {string} "staging" | "development" | "production" - which environment this is called in
+# Notes:
 # Grabs the most recent ci.yaml github workflow run that was triggered from the
 # pull request of the release branch for this version (regardless of whether
 # that run succeeded or failed). The release branch name must be in semver
 # format with a v prepended.
 # This will contain the artifacts we want.
 # https://developer.github.com/v3/actions/workflow-runs/#list-workflow-runs
 get_artifacts_url() {
+  local environment="$1"
+  local branch="$2"
   local artifacts_url
-  local version_branch="v$VERSION"
-  local workflow_runs_url="repos/:owner/:repo/actions/workflows/ci.yaml/runs?event=pull_request&branch=$version_branch"
-  artifacts_url=$(gh api "$workflow_runs_url" | jq -r ".workflow_runs[] | select(.head_branch == \"$version_branch\") | .artifacts_url" | head -n 1)
+  local event
+
+  case $environment in
+    production)
+      # This assumes you're releasing code-server using a branch named `v$VERSION` i.e. `v4.0.1`
+      # If you don't, this will fail.
+      branch="v$VERSION"
+      event="pull_request"
+      ;;
+    staging)
+      branch="main"
+      event="push"
```

Comment:
❓I know `push` is an event. I assume this should _just_ work but only tested briefly. Here's what happens if you call it locally:

```shell
# call from inside code-server repo root
gh api "repos/:owner/:repo/actions/workflows/ci.yaml/runs?event=push&branch=main"
```

response:
```json
{
  "total_count": 526,
  "workflow_runs": [
    {
      "id": 1748197646,
      "name": "Build",
      "node_id": "WFR_kwLOCk8Q9c5oM2EO",
      "head_branch": "main",
      "head_sha": "c20b0b390b54c4d24cda008dfd8041c59fb2f9af",
      "run_number": 4444,
      "event": "push",
      "status": "completed",
      "conclusion": "success",
      "workflow_id": 1216400,
      "check_suite_id": 5062339052,
      "check_suite_node_id": "CS_kwDOCk8Q9c8AAAABLb0p7A",
      "url": "https://api.github.com/repos/coder/code-server/actions/runs/1748197646",
      "html_url": "https://github.com/coder/code-server/actions/runs/1748197646",
      "pull_requests": [
        {
          "url": "https://api.github.com/repos/allinu/code-server/pulls/53",
          "id": 830896171,
          "number": 53,
          "head": {
            "ref": "main",
            "sha": "c20b0b390b54c4d24cda008dfd8041c59fb2f9af",
            "repo": {
              "id": 172953845,
              "url": "https://api.github.com/repos/coder/code-server",
              "name": "code-server"
            }
          },
```